### PR TITLE
ENT-1882: Port PoolResource to spec-first

### PR DESCRIPTION
**Pull request by sonalidhome**
- Added Pool reseaource specification
 - Upated related files

### CODE REVIEW

File: api/candlepin-api-spec.yaml [hunk 1]
```diff
@@ -2849,6 +2849,305 @@ paths:
...
+        - Certificates
+      operationId: getSubCert
+      parameters:
+        - in: path
+          name: pool_id
+          description: Pool ID
+          required: true
+          schema:
+            type: string
+      responses:
```
Comment (wolfdale) [inline api/candlepin-api-spec.yaml:3138]: Missing API response (HTTP 404) when "Pool ID or cert not found".

File: api/candlepin-api-spec.yaml [hunk 2]
```diff
@@ -2849,6 +2849,305 @@ paths:
...
+        200:
+          description: Retrives a single pool by its ID
```
Comment (wolfdale) [inline api/candlepin-api-spec.yaml:2997]: Incorrect description for delete API.

Comment (Januson) [inline api/candlepin-api-spec.yaml:2997]: Wrong description and response code should be 204.

Comment (sonalidhome) [reply api/candlepin-api-spec.yaml]: Corrected.

Comment (sonalidhome) [reply api/candlepin-api-spec.yaml]: Corrected.

File: api/candlepin-api-spec.yaml (hunk 1 above)
Comment (sonalidhome) [reply api/candlepin-api-spec.yaml:3138]: Added.

**Comment by sonalidhome** [comment]
Added break lines in API description.

**Pull request merged**